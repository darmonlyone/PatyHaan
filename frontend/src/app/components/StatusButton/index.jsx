import React from "react";

import usePartyButtonHandle from "app/hooks/usePartyButtonHandle";

const partyStatus = {
	JOINED: "joined",
	OWNED: "owned",
	FULL: "full",
};

const StatusButton = ({ id, status }) => {
	const {
		onJoinHandle,
		onOffPartyHandle,
		onDeleteHandle,
	} = usePartyButtonHandle(id);

	switch (status) {
		case partyStatus.JOINED:
			return (
				<button
					onClick={onOffPartyHandle}
					className="w-3/5 text-sm py-1 bg-blue-400"
				>
					ออกปาร์ตี้
				</button>
			);
		case partyStatus.OWNED:
			return (
				<button
					onClick={onDeleteHandle}
					className="w-3/5 text-sm py-1 bg-red-400"
				>
					ลบปาร์ตี้
				</button>
			);
		case partyStatus.FULL:
			return (
				<button className="w-3/5 text-sm py-1 bg-gray-400" disabled>
					เต็ม
				</button>
			);
		default:
			return (
				<button onClick={onJoinHandle} className="w-3/5 text-sm py-1">
					เข้าร่วม
				</button>
			);
	}
};

export default StatusButton;
