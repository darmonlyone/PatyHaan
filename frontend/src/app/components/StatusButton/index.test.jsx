import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

import StatusButton from ".";

it("should render StatusButton with default value", async () => {
	const { asFragment } = render(
		<Router>
			<StatusButton />
		</Router>
	);

	expect(asFragment()).toMatchSnapshot();
});

it("should render StatusButton with status joined", async () => {
	const { asFragment } = render(
		<Router>
			<StatusButton status="joined" />
		</Router>
	);

	expect(asFragment()).toMatchSnapshot();
});

it("should render StatusButton with status full", async () => {
	const { asFragment } = render(
		<Router>
			<StatusButton status="full" />
		</Router>
	);

	expect(asFragment()).toMatchSnapshot();
});

it("should render StatusButton with default owned", async () => {
	const { asFragment } = render(
		<Router>
			<StatusButton status="owned" />
		</Router>
	);

	expect(asFragment()).toMatchSnapshot();
});
