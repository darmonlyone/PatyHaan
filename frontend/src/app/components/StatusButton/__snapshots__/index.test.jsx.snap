// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render StatusButton with default owned 1`] = `
<DocumentFragment>
  <button
    class="w-3/5 text-sm py-1 bg-red-400"
  >
    ลบปาร์ตี้
  </button>
</DocumentFragment>
`;

exports[`should render StatusButton with default value 1`] = `
<DocumentFragment>
  <button
    class="w-3/5 text-sm py-1"
  >
    เข้าร่วม
  </button>
</DocumentFragment>
`;

exports[`should render StatusButton with status full 1`] = `
<DocumentFragment>
  <button
    class="w-3/5 text-sm py-1 bg-gray-400"
    disabled=""
  >
    เต็ม
  </button>
</DocumentFragment>
`;

exports[`should render StatusButton with status joined 1`] = `
<DocumentFragment>
  <button
    class="w-3/5 text-sm py-1 bg-blue-400"
  >
    ออกปาร์ตี้
  </button>
</DocumentFragment>
`;
